code/game/gamemodes/antag_spawner.dm

```diff
@@ -228,11 +228,14 @@
 
 
 /obj/item/weapon/antag_spawner/slaughter_demon/attack_self(mob/user)
-	var/list/demon_candidates = pollCandidates("Do you want to play as a [initial(demon_type.name)]?", null, null, ROLE_ALIEN, 50)
 	if(user.z != 1)
 		user << "<span class='notice'>You should probably wait until you reach the station.</span>"
 		return
+	var/list/demon_candidates = pollCandidates("Do you want to play as a [initial(demon_type.name)]?", null, null, ROLE_ALIEN, 50)

```

Comment:
might be safer to also check z level after pollCandidates(), maybe the wizard was initially on the station but moved to lavaland or drifted in space during the proc.